mlir/lib/Dialect/MIOpen/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -2222,15 +2221,31 @@ struct GridwiseGemmV2RewritePattern
     auto arrayB = b.create<GpuAllocOp>(loc, arrayBType);
 
     // -----
-
     // Logic to allocate 0-initialized vectors for C.
-    SmallVector<Value, 4> vectorCs;
-    SmallVector<Type, 4> vectorCTypes;
-    auto vectorZeroConst = createZeroConstantOp(b, loc, vectorType);
-    std::fill_n(std::back_inserter(vectorCs), vectorNumber, vectorZeroConst);
-    std::fill_n(std::back_inserter(vectorCTypes), vectorNumber, vectorType);
+    int64_t regCVectorLen = vectorType.getNumElements();
+    Type destType = op.c().getType().cast<MemRefType>().getElementType();
 
-    // -----
+    // Determine the type used on VGPR to act as accumulator.
+    // f32: f32.
+    // f16: f32 to prevent overflow from happening.
+    // i16(bf16) : i16.
+    // i8: i32, since we have an i32 output
+    Type accumulatorType = destType;
+    if (elementType == b.getF16Type() || elementType == b.getBF16Type()) {
+      accumulatorType = b.getF32Type();
+    } else if (elementType == b.getI8Type()) {
+      accumulatorType = b.getI32Type();
+    }
+
+    VectorType accumulatorVectorType =
+        vectorType.cloneWith({}, accumulatorType);
```

Comment:
Is this spelled `clone(accumulatorType)` also?

Reply:
Hmm you are right, it looks like clone is the high level wrapper that allows either shape or type argument.